pytest_twisted.py

```diff
@@ -51,21 +54,75 @@ def stop_twisted_greenlet():
         gr_twisted.switch()
 
 
-def pytest_addhooks(pluginmanager):
+def create_twisted_greenlet():
     global gr_twisted
     if not gr_twisted and not reactor.running:
         gr_twisted = greenlet.greenlet(reactor.run)
         # give me better tracebacks:
         failure.Failure.cleanFailure = lambda self: None
 
 
+def pytest_addhooks(pluginmanager):
+    create_twisted_greenlet()
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup('twisted')
+    group.addoption('--qt5reactor', dest='qt5reactor', action='store_true',
+                    help='prepare for use with qt5reactor')
+
+
+def pytest_configure(config):
+    # TODO: why is the parameter needed?
```

Comment:
Ahh! I think you need to add `self` to `default_reactor` and `qt5_reactor` because you are making them a method of `ReactorPlugin`:

```python
def qt5_reactor(qapp):
```

The `qapp` parameter is being given `self` (which is an instance of `ReactorPlugin`);